### PR TITLE
added split text algorithm and added tests for the new algorithm

### PR DESCRIPTION
**Pull request by itai192**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708&bid=20479097)

File: Algorithms/Strings/GeneralStringAlgorithms.cs [hunk 1]
```diff
@@ -38,5 +38,42 @@ public static class GeneralStringAlgorithms
...
+                if (rem > 0)
+                {
+                    rem--;
+                    times++;
+                }
+
+                string part = string.Empty;
+                while (times != 0)
+                {
+                    part += text[cnt];
```
Comment (siriak) [inline Algorithms/Strings/GeneralStringAlgorithms.cs:67]: Codacy found an issue: [Use a StringBuilder instead.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

File: Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs [hunk 2]
```diff
@@ -20,5 +20,14 @@ public static void MaxCountCharIsObtained(string text, char expectedSymbol, int
             Assert.AreEqual(expectedSymbol, symbol);
             Assert.AreEqual(expectedCount, count);
         }
+
+        [Test]
+        [TestCase("bluh", 4, new string[]{"b","l","u","h"})]
+        [TestCase("fox",2,new string[] {"fo","x" })]
```
Comment (siriak) [inline Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs:26]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

File: Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs [hunk 3]
```diff
@@ -20,5 +20,14 @@ public static void MaxCountCharIsObtained(string text, char expectedSymbol, int
             Assert.AreEqual(expectedSymbol, symbol);
             Assert.AreEqual(expectedCount, count);
         }
+
+        [Test]
+        [TestCase("bluh", 4, new string[]{"b","l","u","h"})]
```
Comment (siriak) [inline Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs:25]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708&bid=20479097)

File: Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs (hunk 2 above)
Comment (siriak) [inline Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs:26]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

File: Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs (hunk 3 above)
Comment (siriak) [inline Algorithms.Tests/Strings/GeneralStringAlgorithmsTests.cs:25]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

File: Algorithms/Strings/GeneralStringAlgorithms.cs [hunk 4]
```diff
@@ -38,5 +38,47 @@ public static class GeneralStringAlgorithms
...
+                if (rem > 0)
+                {
+                    rem--;
+                    times++;
+                }
+                //add the characters to a new string
+                string part = string.Empty;
+                while (times != 0)
+                {
+                    part += text[cnt];
```
Comment (siriak) [inline Algorithms/Strings/GeneralStringAlgorithms.cs:71]: Codacy found an issue: [Use a StringBuilder instead.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6295708)

**Review by siriak — COMMENTED** [review]
Hi, do you have any information as to where this algorithm might be used? Also, references to sources would be good as mentioned in CONTRIBUTING.md. It seems to be too simple to be mentioned somewhere and I don't see the problem it's solving.

**Comment by itai192** [comment]
Hi, I've used this algorithm to split an url into equally destributed parts of text, than used those parts to use in a riddle, I genuinely don't know how useful this algorithm is, but to me it served enough use to consider trying to contribute it to you, and what do you mean sources? I'm new to making pull requests to other repos so it would be nice if you explained it. If this algorithm is to simple for you I'll close the pull request, I just felt it could have been used somewhere else...

**Comment by siriak** [comment]
By sources I mean books, scholarly articles, talks at conferences, webpages on trusted websites like Wikipedia, etc. There are a plethora of algorithms that could be used somewhere, but only some of them are well-studied and solve common problems that come up over and over again, those are what we are looking for. We cannot add lesser-known things because there are so many of them that everything will turn into a big ball of mud and no one will be able to find what they were looking for anyway. Hope that explains my point. Thank you for reaching out anyway and hope to see more well-known algorithms implemented by you! Some of the things we would like to see here you can find in issues.

**Comment by itai192** [comment]
thank you anyway

**Pull request closed without merging**